Allow user to select Kubeconfig from filesystem

ixrock opened the pull request:
![image](https://user-images.githubusercontent.com/6377066/91071365-c1169700-e640-11ea-8515-24f4ac9dc7d5.png)

![image](https://user-images.githubusercontent.com/6377066/91071190-90366200-e640-11ea-916a-8f35ac58e3f8.png)

close #738

ixrock committed:
Allow user to select Kubeconfig from filesystem, fix #738

Signed-off-by: Roman <ixrock@gmail.com>

jakolehm commented on src/renderer/components/input/file-input.tsx:
```diff
@@ -0,0 +1,66 @@
+  render() {
+    const { onSelectFiles, readAsText, ...props } = this.props;
+    return (
+      <input
```
Can we somehow set a directory here? It would make sense to open a file dialog to a path where default kubeconfig is located (thanks @matti for this tip).

ixrock replied:
Have to check if it's possible with standard `<input type="file">` system dialog to customize it somehow.

ixrock replied:
Can't find anything appropriate for https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file

jakolehm replied:
It's not normally allowed because of browser security issues. Electron could have an extension to this behaviour because it also allows to read local files (which browsers don't allow).

jakolehm replied:
Could we use this? https://www.electronjs.org/docs/api/dialog#dialogshowopendialogbrowserwindow-options

aleksfront commented on src/renderer/components/+add-cluster/add-cluster.scss:
```diff
@@ -1,10 +1,34 @@
   }
 
+  .kube-config-select {
+    border: 1px solid $borderColor;
```
```suggestion
    border: 1px solid $borderFaintColor;
```

aleksfront reviewed (commented):
Also, is it possible to show hidden files and folders in upload dialog? It's impossible to go to some paths because of that.

jakolehm reviewed (approved):
LGTM. We probably should open an issue about using electron's `showOpenDialog` (instead of input).

jakolehm commented on src/renderer/components/+add-cluster/add-cluster.tsx:
```diff
@@ -167,12 +178,31 @@ export class AddCluster extends React.Component {
+            htmlFor="kube-config-select"
+            className="kube-config-select flex gaps align-center box grow"
+          >
+            <span className="title">Config file</span>
```
```suggestion
            <span className="title">Kubeconfig file</span>
```

ixrock committed:
Update src/renderer/components/+add-cluster/add-cluster.tsx

Co-authored-by: Jari Kolehmainen <jari.kolehmainen@gmail.com>

ixrock committed:
Update src/renderer/components/+add-cluster/add-cluster.scss

Co-authored-by: Alex Andreev <alex.andreev.email@gmail.com>

ixrock committed:
use `dialog.showOpenDialog` for selecting custom kube-config file

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
enable drag-n-drop for custom kube-config file

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
reworks -- part 1

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
replaced browse/reset buttons to icons, extended drop area for custom kube-config file

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
allow to manually input kube-config file path (edit + save on blur + auto-replace "~" => os.homedir()

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
fixes

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
more fixes

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
fix unit tests

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
fixes:
- remove only custom embed kubeconfig files
- keep link to original kubeconfig file when added via file
- remove auto-select all contexts from kubeconfig source

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
auto-select contexts up to 10, main-layout small fix

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
Merge remote-tracking branch 'origin/master' into add_cluster_from_custom_config

# Conflicts:
#	src/common/cluster-store.ts
#	src/migrations/cluster-store/3.6.0-beta.1.ts

ixrock committed:
master-merge fixes

Signed-off-by: Roman <ixrock@gmail.com>

nevalla commented on src/common/vars.ts:
```diff
@@ -5,7 +5,7 @@ import { defineGlobal } from "./utils/defineGlobal";
 export const isMac = process.platform === "darwin"
 export const isWindows = process.platform === "win32"
-export const isDebugging = process.env.DEBUG === "true";
+export const isDebugging = process.env.DEBUG === "lens";
```
please revert this and let's focus on this in separate PR

ixrock replied:
Done

ixrock committed:
revert `process.env.DEBUG` to "true"

Signed-off-by: Roman <ixrock@gmail.com>

ixrock committed:
remove auto-select clusters, clean up

Signed-off-by: Roman <ixrock@gmail.com>

nevalla reviewed (approved):
LGTM
